facades/PC/src/main/java/org/terasology/engine/Terasology.java

```diff
@@ -120,6 +121,9 @@ public static void main(String[] args) {
             if (!writeSaveGamesEnabled) {
                 CoreRegistry.get(Config.class).getTransients().setWriteSaveGamesEnabled(writeSaveGamesEnabled);
             }
+
+            CoreRegistry.get(Config.class).getNetwork().setServerPort(serverPort);
```

Comment:
This will effectively override the config setting even if the `-serverPort` command line parameter is not set. This should be moved down into the `handleLaunchArguments` method.

But still, changes to `Config` will also be written to file on shutdown, which should not happen. Maybe you can add an entry to `TransientConfig` (those values are not stored permanently) and read from the base Config entry and command line.